role-service/src/test/java/uk/ac/ox/ndph/mts/role_service/controller/RoleControllerTest.java

```diff
@@ -0,0 +1,101 @@
+package uk.ac.ox.ndph.mts.role_service.controller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Test;
+import org.junit.runner.RunWith;
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.result.MockMvcResultHandlers;
+import uk.ac.ox.ndph.mts.role_service.model.Role;
+import uk.ac.ox.ndph.mts.role_service.model.RoleDTO;
+import uk.ac.ox.ndph.mts.role_service.model.RoleRepository;
+
+import javax.ws.rs.core.MediaType;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.springframework.test.web.client.match.MockRestRequestMatchers.content;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(RoleController.class)
+class RoleControllerTest {
+
+
+    @Autowired
+    private MockMvc mvc;
+    @Autowired
+    private ObjectMapper jsonMapper;
+
+    @MockBean
+    private RoleRepository roleRepo;
+
+    @MockBean
+    private ModelMapper modelMapper;
+
+
+    @Test
+    void givenValidRole_whenPost_thenReturnJson()
+            throws Exception {
+
+        String dummyName = "Dummy role name";
+        RoleDTO role = new RoleDTO();
+        role.setRoleName(dummyName);
+
+        String jsonRole = jsonMapper.writeValueAsString(role);
+
+        mvc.perform(post("/roles/")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(jsonRole)
+                .accept(MediaType.APPLICATION_JSON))
+                .andDo(MockMvcResultHandlers.print())
+                .andExpect(status().isOk());
+
+    }
+
+    @Test
+    void givenInvalidRole_whenPost_thenReturn400()
```

Comment:
Do we need to validate that the role name is unique?

Reply:
I am not sure - maybe the ID is the unique property to rely on?   OR if the name is the true unique ID, then perhaps we don't need the numeric ID?   I am in discussion with Adrian about the various properties of roles and permissions today, so I will raise this with him and I'll deal with this in branch 101 if it's to be unique.  OK to leave as-is for this PR?

Reply:
Change of plan - I have confirmed that the name is unique and we can use it as the ID, so I've added the validation and error handling to pick up duplicates.